### PR TITLE
@/CivicSignalBlog: Misc fixes on CivicSignal blog

### PR DESCRIPTION
**Pull request by m453h**
## Description

This PR includes fixes for two issues on the CivicSignal blog;

1. The `FilterBar` for stories is being rendered even when there is just one featured story, this bug was introduced by #983 when resolving the issue of  the `FilterBar` being incorrectly rendered too close to the `FeaturedArticle`. 

2. Site Header and Footer content not being loaded due to the changed name of the globals configuration slug from `settings site` to `settings-research-site`

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

## Screenshots
Before the fixes
Screenshot 2024-11-22 at 12 31 18

After the fixes
Screenshot 2024-11-22 at 12 28 37

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: apps/civicsignalblog/migrations/20241122_054517_migration.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+export async function up({ payload }: MigrateUpArgs): Promise<void> {
+  await mongoose.connect(mongoURL);
+  const db = mongoose.connection.db;
+
+  const existingGlobalConfig = await db.collection("globals").findOne({
+    globalType: oldGlobalConfigType,
+  });
+
+  if (!existingGlobalConfig) {
```
Comment (kilemensi) [inline apps/civicsignalblog/migrations/20241122_054517_migration.ts:16]:
```suggestion
  const existingOldGlobalConfig = await db.collection("globals").findOne({
    globalType: oldGlobalConfigType,
  });

  if (!existingOldGlobalConfig) {
```

File: apps/civicsignalblog/migrations/20241122_054517_migration.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    payload.logger.error(
+      `No existing global settings [${oldGlobalConfigType}] config found, exiting...`,
+    );
+    return;
+  }
```
Comment (kilemensi) [inline apps/civicsignalblog/migrations/20241122_054517_migration.ts:21]: Should this **really** be an error (`payload.logger.error`)?

File: apps/civicsignalblog/migrations/20241122_054517_migration.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  let newGlobalConfig = await db.collection("globals").findOne({
+    globalType: newGlobalConfigType,
+  });
+
+  if (newGlobalConfig) {
+    payload.logger.error(
+      `Migration failed: Unable to migrate [${oldGlobalConfigType}] to [${newGlobalConfigType}] because a global configuration with this name already exists.`,
+    );
+    return;
+  }
```
Comment (kilemensi) [inline apps/civicsignalblog/migrations/20241122_054517_migration.ts:32]:
```suggestion
  const existingNewGlobalConfig = await db.collection("globals").findOne({
    globalType: newGlobalConfigType,
  });

  if (existingNewGlobalConfig) {
    payload.logger.error(
      `Migration 20241122_054517 failed: Global type [${newGlobalConfigType}] already exists.`,
    );
    return;
  }
```

Just used `20241122_054517` from filename as migration name but if there is a better name, lets use that.

File: apps/civicsignalblog/migrations/20241122_054517_migration.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+  try {
+    newGlobalConfig = {
+      ...existingGlobalConfig,
+      globalType: newGlobalConfigType,
+    };
+
+    delete newGlobalConfig._id;
```
Comment (kilemensi) [inline apps/civicsignalblog/migrations/20241122_054517_migration.ts:40]:
```suggestion
    const { _id, globalType, ...newGlobalConfig } = existingOldConfig;
    newGlobalConfig.globalType = newGlobalConfigType;
```

**Review by kilemensi — COMMENTED** [review]
👍🏽 

---

1. Lets prefer using `const` over `let`.
2. Is the `ArticleGrid` component in this app different from the one in [`codeforafrica`](https://github.com/CodeForAfrica/ui/blob/main/apps/codeforafrica/src/components/ArticleGrid/ArticleGrid.js) visible [here](https://cfa.dev.codeforafrica.org/stories)?

**Comment by m453h** [comment]
> 👍🏽
> 
> 1. Lets prefer using `const` over `let`.
> 2. Is the `ArticleGrid` component in this app different from the one in [`codeforafrica`](https://github.com/CodeForAfrica/ui/blob/main/apps/codeforafrica/src/components/ArticleGrid/ArticleGrid.js) visible [here](https://cfa.dev.codeforafrica.org/stories)?

Yes this is exactly the same component, I've replaced the current implementation in this PR with what we have on `codeforafrica` it does solve the original problem we had in the `civicsignalblog` which was being rendered as 👇 
Screenshot 2024-11-22 at 13 26 49

Instead of:
Screenshot 2024-11-22 at 13 28 33


Only one problem that I will push a fix for both `codeforafrica` and `codeforafrica` is the case of when we are only having one story, we shouldn't render the filter bar 👇 
Screenshot 2024-11-22 at 13 30 49

File: apps/civicsignalblog/migrations/20241122_054517_migration.ts (hunk 3 above)
Comment (m453h) [reply apps/civicsignalblog/migrations/20241122_054517_migration.ts:32]: Filename looks good, we can easily trace which migration failed👍

File: apps/civicsignalblog/migrations/20241122_054517_migration.ts (hunk 2 above)
Comment (m453h) [reply apps/civicsignalblog/migrations/20241122_054517_migration.ts:21]: Hmm, we can keep it as `info` 🤔

File: apps/civicsignalblog/migrations/20241122_054517_migration.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+  try {
+    const { _id, globalType, ...newGlobalConfig } = existingOldGlobalConfig;
+    newGlobalConfig.globalType = newGlobalConfigType;
+    await db.collection("globals").insertOne(newGlobalConfig);
+    payload.logger.info(
+      `✓ Successfully migrated global config [${oldGlobalConfigType}] to [${newGlobalConfigType}]`,
+    );
+  } catch (error) {
+    payload.logger.error("Error: Migration failed.");
```
Comment (kilemensi) [inline apps/civicsignalblog/migrations/20241122_054517_migration.ts:42]:
```suggestion
    payload.logger.error("Error: Migration 20241122_054517 failed.");
```

**Review by kilemensi — APPROVED** [review]

**Pull request merged**